docs/content/latest/explore/ysql-language-features/gin.md

```diff
@@ -0,0 +1,501 @@
+# Generalized inverted indexes
+
+In YugabyteDB, tables and secondary indexes are both key-value stores.
+For tables, the key-value store maps primary keys to values.
+For secondary indexes, the key-value store maps index keys to primary keys.
+
+**Regular indexes index columns.**
+This makes queries with conditions on the columns more efficient.
+For example, if one had a regular index on a single int array column (currently not possible in Yugabyte), queries like `WHERE myintarray = '{1,3,6}'` would be more efficient when using the index.
+However, queries like `WHERE myintarray ? 3` (meaning "is 3 an element?") would not benefit from the regular index.
+
+**Generalized inverted indexes (GIN indexes) index elements inside container columns.**
+This makes queries with conditions on elements inside the columns more efficient.
+The above example would benefit from a GIN index since we can look up the key `3` in the gin index.
+
+## Compatible types
+
+GIN indexes can only be created over a few types:
+
+- a GIN index on a tsvector column indexes text elements
+- a GIN index on a array column indexes array elements
+- a GIN index on a jsonb column indexes keys/values
+
+With extensions, more types can be supported.
+However, extension support is still in progress:
+
+- [ ] btree_gin
+- [ ] hstore
+- [x] pg_trgm
```

Comment:
this does not look right in the preview:
![image](https://user-images.githubusercontent.com/4038003/142069461-0d792580-b85a-41cb-a73d-54a44fd38305.png)